src/Chainweb/SPV.hs

```diff
@@ -185,12 +190,31 @@ proofChainId = to (\(TransactionProof cid _) -> cid)
 
 -- -------------------------------------------------------------------------- --
 -- Output Proofs
+--
+-- | Targets for proofs, usually a chain ID, but in the case of L2, a network
+-- and chain identifier. TODO: determine format, if any.
+data ProofTarget = ProofTargetChain !ChainId | ProofTargetCrossNetwork !Text
+    deriving stock (Show, Eq, Ord, Generic)
+    deriving anyclass NFData
```

Comment:
Let's talk about this first before spending too much time on the implementation. I don't yet fully understand this design and I am not yet convinced that it makes sense. This changes the API surface of SPV and any change we do here we have to support forever.